### DISCUSSION TITLE
I want to set specific specs to run in each machine with cypress parallelization in azure-pipeline

### DISCUSSION THREAD

**Discussion by Mturki94** [cypress-io/cypress#15429] [CI setup] [answered]
- I'm using azure-pipelines CI to run my cypress tests, I runs my files in parallelization on two different machines at the same time to decrease the total time, Cypress parallelization divides the files **depending on the time each one take**, But i want to divide the files on each machine **depending on my selection** (for ex. each folder on different machine), How can i do that?

**Comment by bahmutov** [comment, marked as answer]
Unfortunately, this is not possible. Of course you can always just specify the specs to run yourself and just use the same CI build ID to link these specs into the single logical recording on the Dashboard

```
# first machine
npx cypress run --record --ci-build-id $SOME_BUILD_ID --spec cypress/integration/b.js,cypress/integration/c.js
# second machine
npx cypress run --record --ci-build-id $SOME_BUILD_ID --spec cypress/integration/a.js,cypress/integration/d.js
```

Or you can use https://on.cypress.io/module-api to execute `cypress.run` from your code and apply any logic you want

**Reply by Mturki94** [reply]
Thank you for replay, But i can't understand yet how to pass each command for each machine in the example you write?
